tests/functional/apps/zilliqa.py

```diff
@@ -67,6 +68,17 @@ def send_async_get_public_key(self, index: int,
                                   disp_addr: bool) -> Generator[None, None, None]:
         p1 = 0
         p2 = P2_DISPLAY_ADDRESS if disp_addr else P2_DISPLAY_PUBKEY
+
+        payload = pack("<I", index)
+        with self._backend.exchange_async(CLA, INS.INS_GET_PUBLIC_KEY,
+                                          p1, p2, payload):
+            yield
+
+    @contextmanager
+    def send_get_public_key_non_confirm(self, index: int) -> Generator[None, None, None]:
+        p1 = 0
+        p2 = P2_DISPLAY_NONE
+
         payload = pack("<I", index)
         with self._backend.exchange_async(CLA, INS.INS_GET_PUBLIC_KEY,
                                           p1, p2, payload):
```

Comment:
You should use a sync call here as there is no user interaction that should be done.
So it can be:
```suggestion
    def send_get_public_key_non_confirm(self, index: int) -> RAPDU:
        p1 = 0
        p2 = P2_DISPLAY_NONE

        payload = pack("<I", index)
        return self._client.exchange(CLA, INS.INS_GET_PUBLIC_KEY,
                                     p1, p2, payload):
```